### ISSUE TITLE
Feature request: TODO lists within cards

### ISSUE THREAD

**Issue by smarek** [wekan/wekan#597]
Hi,

wekan is really great, one feature from similar kanban apps what I'm missing is TODO lists within cards.
- [x] in menu to card, add option "add todo list"
- [x] todo list is named
- [x] single items in todo list can be checked or unchecked (done status)
- [ ] links in list items should be automatically recognized and clickable
- [ ] When finished input of checklist item description, enter should directly open new checklist item
- [ ] Reordering of checklist items via drag&drop would be great feature
- [x] in overview of board, number of available and done items in todo/check lists should be displayed, ie. like this ![2016-05-08-222220_69x26_scrot](https://cloud.githubusercontent.com/assets/384824/15100197/845c5674-156b-11e6-9a9a-f328deb1c671.png)

Do you think this could be done?

**Update PR #630 from @lkisme is working great, updated checklist here according to his PR state**

**Comment by deanpcmad** [comment]
+1 - I've got a couple of boards on Trello that I'd like to move over to my self hosted Wekan install and this is holding me back. Would love to do list support :)

**Comment by wnereiz** [comment]
+1
Please add the feature. It's really useful. Thanks!

**Comment by irfanitw** [comment]
+100 this is huuuuge for us. Without this, I can't even ask somebody to move from Trello to Wekan

**Comment by irfanitw** [comment]
Looks like this is already being implemented in the next version: https://github.com/wekan/wekan/issues/298

**Comment by AlexanderS** [comment]
@irfanitw #298 is not closed. This is still an open bug report.

**Comment by brylie** [comment]
The PR #576 provides a working todo list feature. We are having difficulty tracking down the PR author, @andersonlin, so they can sign the Wekan contributor agreement. We are also awaiting feedback from @mquandalle regarding issues #129, #298, and this latest issue #597.

**Comment by irfanitw** [comment]
@brylie thanks for connecting the dots for us. I see the PR has been outstanding since April 20th. Is there no way around this? My guess if @andersonlin hasn't PRed it and signed the agreement since April 20th, we might be waiting on a last cause. Altho I do see he was last active 20 days ago in another project: https://github.com/andersonlin?tab=activity

**Comment by brylie** [comment]
Another approach might be to loosen the requirement for contributors to sign a contributor's agreement. @mquandalle any thoughts on how we can move forward on PR #576?

**Comment by smarek** [comment]
Is it possible to reparent the PR and have the contributor's agreement signed by owner of new PR ? Would it be possible, ie. if I got written agreement with @andersonlin that I own the rights to the code?

**Comment by brylie** [comment]
@smarek that is a good idea. How would you document the agreement?

**Comment by smarek** [comment]
@brylie that was the question, i didn't read the CLA of this project in the first place. My idea was to have it confirmed from e-mail address linked with the original PR commits

**Comment by irfanitw** [comment]
@brylie, @smarek, another suggestion could be for another developer to use @andersonlin's code as inspiration. Not exactly copy it line for line, but use it as a guide.

The reason I'm pushing on this pretty hard is this is the one feature that is blocking our Trello users from moving to Wekan. Also, I'm sure their next request is multiple checklists in a card, but one step at a time!

**Comment by lkisme** [comment]
@irfanitw I tried to use the PR, yet there are some features my team finds them not so reasonable, such as only the creator of a specific "check list item" can check it as finished, I'm actually planning make a new implementation on that if there isn't someone else is doing that

**Comment by smarek** [comment]
@lkisme that would be best, if you could provide your own PR with checklist functionality, can you please take a guess on ETA ? Thank you

**Comment by irfanitw** [comment]
@lkisme you're absolutely right, anybody should be able to make a checklist item as completed. Both Wekan and Trello don't have the sense of "ownership" and anybody can be a "member" of a card, so you have the right logic!

**Comment by lkisme** [comment]
I think it would take 1 or 2 weeks to implement that, given the design in #576 can be referred to

**Comment by danger89** [comment]
Still not possible?

**Comment by irfanitw** [comment]
@lkisme how's it going on this?

**Comment by lkisme** [comment]
Will make a PR on this in days

**Comment by lkisme** [comment]
@urbanhop 
You seem to wanna build the project, did you do that properly? I see you didn't use the option of --directory.

I suggest you try to run the program by command "MONGO_URL=mongodb://localhost:27017/wekan meteor" in the project root directory after the diff of #630 is merged, the feature should be there, otherwise please check if the collection of "checklists" within the mongo has been created.

**Comment by urbanhop** [comment]
thx, your are right, the directory switch is necessary.

unfortunately meteor build simply crashes 99% of the time (for instance "fibers" issue, a well known problem. and this on a blank free linux machine, on osx and on win) so i have to give up.

nice work you did, too bad meteor being such a *** basis of wekan. would be great to have such a client app based on a working framework.

**Comment by AlexanderS** [comment]
`meteor build` always works for me. What versions of node and meteor you are using?

**Comment by urbanhop** [comment]
1.3.4.1 (current) problem is the fibers module

**Comment by AlexanderS** [comment]
Wekan is currently using meteor 1.3.2.4. You have to upgrade it first or use the old version of meteor. As stated [here](https://github.com/meteor/meteor/issues/7302#issuecomment-228982819) errors in the `fibers` module are common, but the it's simply where the error is being caught and reported.

**Comment by urbanhop** [comment]
thx for your assistance.

after a long way i built with 1.3.2.4 (i hope so, getting a certain meteor version is not straightforward either) and the new effect is:

    node main.js

nothing happens. is there a way to get output ? some verbose flag in node?

**Comment by urbanhop** [comment]
Getting an older version of Meteor has issues in its own way, I managed to use 

    meteor --release 1.3.2.4

with every command, including help.

After a fresh build

    node main.js

causes a hanging application. (have not seen that before). How can I get some output? Is there some verbose flag?

thx

**Comment by urbanhop** [comment]
dev mode works, checklists are great,
build does not work, which node version should I use? is there any documentation on meteor web page about which node version one is allowed to use?

**Comment by AlexanderS** [comment]
Meteor currently requires node 0.10.x. Once meteor-1.4 is released it will support node-4.4.x.

**Comment by urbanhop** [comment]
got it working, thx for help.

checklist implementation is great, very important feature, thank you so much for your work lkisme.

(reordering of checklist-items would be a nice to have, maybe via drag and drop.)

**Comment by brylie** [comment]
Ping @mquandalle. Any timeline for a PR review?

**Comment by spacekookie** [comment]
It's been a few months. I just recently started using wekan and I'd be very interested in this feature.

Could somebody from the team maybe put an ETA or priority on this issue? As support for check lists is very vital for how I work :(

Thank you!

**Comment by urbanhop** [comment]
lkisme has already done a really nice implementation, which you can fetch it from here forked repo. even detects and visualizes completion. only thing i wished for is reordering of task items.

and certainly this should be pulled here.

**Comment by spacekookie** [comment]
> lkisme has already done a really nice implementation, which you can fetch it from here forked repo. even detects and visualizes completion. only thing i wished for is reordering of task items.

I would much rather not fiddle with our production setup to get a feature off a fork. But it seems like I won't have a choice if this doesn't get pulled in soon...

**Comment by wnereiz** [comment]
Please merge this function. It is important for us. It's the only reason prevent us from migrating to wekan.

**Comment by brylie** [comment]
Please also participate in the 'Wekan community governance' discussion.
https://github.com/wekan/wekan/issues/640

Our goal is to have more of the Wekan community members able to manage the Wekan organization, including reviewing/merging PRs such as this.

**Comment by xet7** [comment]
This issues has gone offtopic. Some actual issue is continued at #828 

If you don't find some actual feature request already open here, open new issue.

Closing this issue.
